src/Company/Company.ts

```diff
@@ -53,26 +37,25 @@ export class Company {
    * For example, the base stat requirement for an intern position is 1.
    * But if a company has a offset of 200, then you would need stat(s) of 201
    */
-  jobStatReqOffset: number;
+  jobStatReqOffset = 0;
 
-  /** Properties to track the player's progress in this company */
-  isPlayerEmployed: boolean;
-  playerReputation: number;
-  favor: number;
+  // Dynamic info, loaded from save and updated during game.
+  playerReputation = 0;
+  favor = 0;
 
-  constructor(p: IConstructorParams = DefaultConstructorParams) {
+  constructor(p?: IConstructorParams) {
+    if (!p) return;
     this.name = p.name;
     this.info = p.info;
-    this.companyPositions = p.companyPositions;
+    p.companyPositions.forEach((jobName) => (this.companyPositions[jobName] = true));
     this.expMultiplier = p.expMultiplier;
     this.salaryMultiplier = p.salaryMultiplier;
     this.jobStatReqOffset = p.jobStatReqOffset;
 
-    this.isPlayerEmployed = false;
     this.playerReputation = 1;
     this.favor = 0;
-    this.isMegacorp = false;
-    if (p.isMegacorp) this.isMegacorp = true;
+    this.hasFaction = false;
+    if (p.hasFaction) this.hasFaction = true;
```

Comment:
```suggestion
    this.hasFaction = !!p.hasFaction;
```

Reply:
Yeah I hadn't really messed with this code, it just looked like modified code because of a variable rename (isMegaCorp was a bad name, especially with a company/faction actually named Megacorp in the game). In progress commit addresses this.